### PR TITLE
Answer:1 Projection

### PR DESCRIPTION
**Pull request by Brack0**
## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

## Notes

- Missing some types on `let-item` but there is challenge 4 for this one 😅
- TemplateContext with `$implicit` is a nice trick

### CODE REVIEW

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
+    this.addItem.emit();
+  }
+
+  trackById(_index: number, item: ListItem) {
+    return item.id;
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:53]: in a few days, you will be able to use the new control flow with the new track function 👍 but nice to have implement it, not everyone think of it.

File: apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts [hunk 2]
```diff
@@ -1,35 +1,60 @@
...
       }
     `,
   ],
   standalone: true,
-  imports: [CardComponent],
+  imports: [
+    CardComponent,
+    ListItemComponent,
+    ListItemTemplateDirective,
+    AsyncPipe,
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts:40]: you can use the new signal API to get read of this. Signal will be the new way of binding properties to the template, it's better to get use to it now. And many benefits will come in a very near future

**Comment by tomalaforge** [comment]
very nice work. you thought of everything. Maybe refacto to signals to make it perfect. 

And as your said, you could have type check the directive. But there is a challenge for that. 😉

File: apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts (hunk 2 above)
Comment (Brack0) [reply apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts:40]: Yep, still in developer preview (as of v16) but could make use of it. Hoping that interop with RxJS is working as intented.

**Comment by Brack0** [comment]
Cya for the next challenge !

**Pull request closed without merging**